lib/rucio/core/did.py

```diff
@@ -43,12 +43,13 @@
 
 if TYPE_CHECKING:
     from collections.abc import Callable, Sequence
-    from typing import Any, Optional, Union
+    from typing import Any, Dict, Optional, Union, List
     from sqlalchemy.orm import Session
     from sqlalchemy.schema import Table
     from rucio.common.types import InternalAccount, InternalScope
 
     LoggerFunction = Callable[..., Any]
+    DIDListType = List[Dict]
```

Comment:
```suggestion
```

I don't think it's worth defining `list[dict]` as a separate type, as it's still fairly generic. If you want, you can create a `TypedDict` that matches the `did` information (e.g. `DIDDict` and then use `list[DIDDict]` in the typing below).